### DISCUSSION TITLE
KRATOSxGiD for Structure response against Blast Load

### DISCUSSION THREAD

**Discussion by engrjessie** [KratosMultiphysics/Kratos#10965] [Q&A]
Hello everyone, I am a newbie here. I am doing a study about blast load on masonry structures using FEA. I am really desperate now to reach out to anyone with a huge knowledge about this amazing  software since I think this could be a great help in my current graduate study about blast loading (CFD) effect on masonry structure (structure). I hope you could have the time to answer my question because I will greatly appreciate it. 

My question is, I have already installed GiD simulation and loaded already the Kratos problem data type, is KRATOSxGiD capable of calculating the response of the structure including cracks and failure due to an explosion due to shock and pressure (like the FSI function)? If yes, is there any way for me to study or any tutorial that would be a great help. Or even webinars is an amazing opportunity for me.

**Comment by rubenzorrilla** [comment]
Dear @engrjessie,

Kratos implements both weakly-compressible and compressible (in conservative form) Navier-Stokes formulations. One of these might suit your target applications. In the past we have used the weakly compressible approach for the simulation of low impedance failures in electric transformers.
About the mechanical problem, you need to provide more information on how would you target the masonry problem (damage claw, RVE, etc...). Depending on this, the technology would be more or less mature. As far as I know, only the damage constitutive models are production ready, the rest of technologies are ongoing research lines.

Having said this, I also note that all of these capabilities are not present in the GID GUI, so using them would require you at least Python scripting in order to build your model up. I also note that this is an extremely challenging problem that would require you an advance training in Kratos. Regarding the tutorials, unfortunately they are much basic than the problem you aim at solving.

**Reply by engrjessie** [reply]
So to clarify your answer,basically KRATOS is very much capable to do the task in order for me to obtain my objective (to investigate and simulate response and failure of masonry structure with internal blast)? I just want to ask on what would be the first step that I should take in order to have an advance knowledge or training using Kratos? I really wanted to draft a conceptual framework for this study in order for me to contribute to the kratos community.

**Reply by rubenzorrilla** [reply]
I'd say that Kratos is very much capable to be the framework on top of which you can build your developments to achieve your objective. Depending on the formulations and methods you require, you would need to develop more or less stuff. As I said above, we need you to be more specific in this regard (formulation, discretization, numerical techniques, etc you want to use) in order to tell you if the stuff you require is already implemented or not.

**Reply by engrjessie** [reply]
The idea that is on my mind is basically two parts. The first part is that that an explosive charge (say TNT) inside a structure will be detonated that will cause shockwave and gas pressure inside the structure, so I need to determine the pressure it exerts to the confining wall (CFD). Then the second part would be the response (strain and stresses) of the walls including cracks or fractures (if there are any, due to the high pressures) using FEA. So basically, I wanted to couple this two procedure, CFDxFEA in order to have a more exact simulation and result which results to a dynamic remeshing or anything like this.

**Reply by rubenzorrilla** [reply]
This is clear. However, in order to provide some guidance we need you to be more specific about the technologies you want to use. For instance, depending on the flow regime generated by the explosion a weakly-compressible form might be sufficient. This is related to how the explosive charge is modelled.  About the mechanical response, what approach are you planning to use for the wall modelling?, is a damage model sufficient or you really  need to model the cracks? On top of this there is the coupling strategy (implicit or explicit), which will largely depend on the approach you want to use for the CFD domain.

As I said before, this is a really challenging problem which requires to carefully consider the simulation strategy to be applied. Once you have this clear, we can ping the experts on each topic to guide you on what is already donde and what would be required to be done.

**Reply by engrjessie** [reply]
I was reading the available literature and most of them is using the Navier-Stokes as the governing equation and BKW equation of state for the Fluid due to the explosion. And for the structural response, I have been using the usual FEM formulation (see below for the stiffness matrix, k) but I am not quite sure on what would be the additional formulation so that I can model the damage (damaged walls or failures) due to separation using nonlinear analysis. That is why I would like to clarify if I could do additional scripting on KRATOS for this. I hope I have answered your question regarding my concern.

![image](https://user-images.githubusercontent.com/129816555/230273790-a2a9d01c-1554-465f-a7bd-47c05462efc9.png)

**Reply by engrjessie** [reply]
I want to share one of my small activity using FEM on a RC structure due to a fictitious ground motion. My goal actually is almost similar to this but the difference is that the load is due to the pressure from the explosion, and the analysis should simulate fracture or even separation of the elements due to the overpeak pressure that will cause brittle failure to the structure. 

https://761vqp-my.sharepoint.com/:b:/g/personal/jessiecorralesb_761vqp_onmicrosoft_com/EZZXqqBlgNlIjRyMnbn25jsBBsz51c9f_9hBIEYyKdrF5w?e=tb5cli

https://761vqp-my.sharepoint.com/:p:/g/personal/jessiecorralesb_761vqp_onmicrosoft_com/EVGfKewjXe9KpVko7jrvwH0BqT7K2SYPoG6djfu96J5tEA?e=HuBgzf

I hope this would make my objective more clearer. Thank you very much.

**Reply by engrjessie** [reply]
In summary, I was aiming for the Arbitrary Lagrangian- Eulerian (ALE) formulation for the FSI model where Eulerian formulation for the Navier- Stokes equations on the fluid side while Lagrangian formulation on the structure side. And the flow of the analysis would be somewhat like the same as below a cycle for everytime step.

![image](https://user-images.githubusercontent.com/129816555/230290265-671b39bb-0b51-4d8f-b756-4be649eeac7e.png)

**Reply by engrjessie** [reply]
https://vimeo.com/50838557?embedded=true&source=vimeo_logo&owner=5139387

I saw this introductory video of KRATOS in the webpage and my expected output is shown in timestep 1:12 to 1:36 which is all about damage and collapse.

**Comment by rubenzorrilla** [comment]
I'm not aware of this BKW equation of state. Our compressible formulation hardcodes the ideal gas equation, so if you are to use a compressible form + BKW EOS you'll need to implement it. In the past, we did something similar with a weakly-compressible form and the Tait EOS, you can find the details in [here](https://upcommons.upc.edu/handle/2117/360346). I'm not sure if this approach is suitable for your explosion regime, but if it is, it would definitively require a much easier implementation (indeed, only Python scripting).

Regarding the structure, you need to still define the target of the simulation and the model behind. If you're only interested on checking if there is failure or not, a damage model can be sufficient. If you're interested in the separation of the elements, this would require a much more complicated numerical technique. In any case, I ping my colleague  @AlejandroCornejo who has a vast experience in this regard.

Finally, about the coupling, the strategy (explicit or implicit) would largely depend on the subdomain solvers one. So you first need to define these.

**Reply by engrjessie** [reply]
Thank you sir for this very useful insights, I am still reviewing availble literatures so I still need to understand more of the concept so that I could be more specific with this one.  My ultimate goal is really on the separation of elements using FEA, I know this is would be a complicated one but its worth a try maybe.

**Reply by engrjessie** [reply]
I have read the article about the Tait EOS you used with the weakly compressible form, I think this would be a a good reference for my study. With regards to this in depth learning with KRATOS, especially on the scripting, would this be appropraite for an MS thesis? especially on the code formulation for element separation. I am currently starting to explore the KRATOS in GiD so that I can be familiarized with the functions.

**Reply by rubenzorrilla** [reply]
I think it would be. Indeed the reference I attached is a MSc thesis.

**Reply by AlejandroCornejo** [reply]
Hi @engrjessie ! I assume that you contacted me before on YouTube?  :)

As @rubenzorrilla  says, the coupling scripting is not a big deal if you know what to customize. However, I am curious about the "element separation" thta you're mentioning. Do you want to detach elements by duplicating nodes? How are you going to model contact between elements? In my thesis I was removing elements to create cracks instead of separating elements [(link to my thesis)](https://www.researchgate.net/publication/347934285_A_fully_Lagrangian_formulation_for_fluid-structure_interaction_between_free-surface_flows_and_multi-fracturing_solids)

**Reply by engrjessie** [reply]
Hi Sir, @AlejandroCornejo, yes sir. Ive actually read all your articles related to my study. But some of your paper uses FEM/DEM coupling to simulate large deformations.  The element speration that i was mentioning is something like the spalling of masonry elements during simulations, but i think as ive read the literature this would be a complex matter. So if in any case, this "spalling simulation" is not feasible for the time being, i am considering the erosion algorithm (if that can be done in KRATOS) just to simulate the failure of a structure by removing failed elements or something like this.

PS: I appreciate so much for your help and insights sir @rubenzorrilla and @AlejandroCornejo , its an honor for me.

**Reply by engrjessie** [reply]
@AlejandroCornejo I think I have read te article about this disserttation of yours sir. I think my question was answered already sir base on your discussion in chapter 22 since it was related to blast. This is actually a much better procedure compare to the one that I am thinking ( i am not quite sure if your process sir is somewhat like the erosion algorithm in LS-DYNA).
